src/expr/node_algorithm.cpp

```diff
@@ -304,6 +304,48 @@ void getSymbols(TNode n,
   } while (!visit.empty());
 }
 
+void getOperatorsMap(TNode n, std::map<TypeNode, std::unordered_set<Node, NodeHashFunction>>& ops)
+{
+  std::unordered_set<TNode, TNodeHashFunction> visited;
+  getOperatorsMap(n, ops, visited);
+}
+
+void getOperatorsMap(TNode n,
+                     std::map<TypeNode, std::unordered_set<Node, NodeHashFunction>>& ops,
+                     std::unordered_set<TNode, TNodeHashFunction>& visited)
+{ 
+  //nodes that we still need to visit
+  std::vector<TNode> visit;
+  //current node
+  TNode cur;
+  visit.push_back(n);
+  do
+  {
+    cur = visit.back();
+    visit.pop_back();
+    //if cur is in the cache, do nothing
+    if (visited.find(cur) == visited.end())
+    {
+      //fetch the correct type and add it to the resulting map if needed
+      TypeNode tn = cur.getType();
+      if (ops.find(tn) == ops.end())
+      {
+        ops[tn] = std::unordered_set<Node, NodeHashFunction>();
+      }
```

Comment:
This whole if statement above is not needed. When you do `ops[tn].insert(NodeManager::currentNM()->operatorOf(cur.getKind()))` below, the `std::unordered_set` will automatically be created when `ops[tn]` is executed.